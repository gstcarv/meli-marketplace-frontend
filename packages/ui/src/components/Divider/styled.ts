import styled from 'styled-components';

export const Container = styled.main`
    background-color: #ffffff;
    margin: 0 10%;
    margin-top: ${({ theme }) => theme.spacing.get('xl')};
    padding: ${({ theme }) => theme.spacing.get('md')};
    border-radius: ${({ theme }) => theme.radii.get('xs')};
`;
