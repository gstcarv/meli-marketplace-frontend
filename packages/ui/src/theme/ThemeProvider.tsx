import 'react-loading-skeleton/dist/skeleton.css';
import { PropsWithChildren } from 'react';
import { ThemeProvider as StyledThemeProvider } from 'styled-components';
import { GlobalStyle } from '../globalStyles';
import { theme } from './theme';

export const ThemeProvider = (props: PropsWithChildren) => (
    <StyledThemeProvider theme={theme}>
        <>
            {props.children}

            <GlobalStyle />
        </>
    </StyledThemeProvider>
);
