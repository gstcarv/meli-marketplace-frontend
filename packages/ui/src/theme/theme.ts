import { colors, fontSize, radii, spacing, fontWeight } from '../tokens';

export const theme = { colors, fontSize, radii, spacing, fontWeight };

export type Theme = typeof theme;
