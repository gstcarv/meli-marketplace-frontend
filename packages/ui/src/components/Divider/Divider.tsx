import { Spacing } from '@meli/design-tokens';
import styled from 'styled-components';

type Props = {
    my?: Spacing;
};

export const Divider = styled.hr<Props>`
    border: 0;
    height: 1px;
    background: ${({ theme }) => theme.colors.get('gray-100')};
    width: 100%;
    margin: ${({ my, theme }) => (my ? theme.spacing.get(my) : 0)} 0;
`;
