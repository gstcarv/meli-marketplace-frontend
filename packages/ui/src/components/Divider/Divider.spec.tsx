import { render } from '@testing-library/react';
import { ThemeProvider } from '../../theme';
import { Divider } from './Divider';

describe('<Divider />', () => {
    it('should render correctly', () => {
        const { container } = render(
            <ThemeProvider>
                <Divider />
            </ThemeProvider>
        );

        expect(container).toMatchSnapshot();
    });
});
