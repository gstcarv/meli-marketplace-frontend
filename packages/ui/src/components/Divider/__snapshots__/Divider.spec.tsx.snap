// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Divider /> should render correctly 1`] = `
<div>
  <hr
    class="sc-blHHSb OFIbg"
  />
</div>
`;
